### PR TITLE
README: add tools install instructions

### PR DESCRIPTION
**Pull request by pro-wh**
and people make fun of npm projects for having devDependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,29 @@ $ curl -X GET http://localhost:8008/v1
...
+curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
+```
+
+**gofumpt**: [go install](https://pkg.go.dev/mvdan.cc/gofumpt)
+
+```sh
+go install mvdan.cc/gofumpt@latest
+```
+
+**goimports**: [go installl](https://pkg.go.dev/golang.org/x/tools/cmd/goimports)
```
Comment (mitjat) [inline README.md:104]: Nit: Triple L

File: README.md [hunk 2]
```diff
@@ -84,6 +84,29 @@ $ curl -X GET http://localhost:8008/v1
...
+
+```sh
+go install golang.org/x/tools/cmd/goimports@latest
```
Comment (mitjat) [inline README.md:107]: `latest` always makes me nervous. These are just manual instructions, so I guess it's fine ... Your call.
FWIW, I have
```
$ go list -m -json golang.org/x/tools  # contains goimports
{
        "Path": "golang.org/x/tools",
        "Version": "v0.11.0",
        "Time": "2023-07-05T15:50:13Z",
        "Indirect": true
}

$ gofumpt -version
v0.4.0 (go1.19.1)
```
and CI uses [golangci-lint v1.49](https://github.com/oasisprotocol/nexus/blob/2e858695add9655b1b4bedeb67c0aee0bb64a815/.github/workflows/ci-lint.yaml#L51-L51)

File: README.md [hunk 3]
```diff
@@ -84,6 +84,29 @@ $ curl -X GET http://localhost:8008/v1
...
 
+### Code quality tools
+
+Here are our recommendations for getting the tools that `make lint` and `make fmt` use.
+
+**golangci-lint**: [use their `curl | sh` installer](https://golangci-lint.run/usage/install/).
+They explain that the tool is in dependency hell and any more structured distribution of it might not work.
+
+```sh
+curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
```
Comment (mitjat) [inline README.md:95]:
```suggestion
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.49.0
```

to match CI

**Review by mitjat — APPROVED** [review]
Thank you! :pray:

File: README.md (hunk 1 above)
Comment (pro-wh) [reply README.md:104]: :scream:

File: README.md (hunk 3 above)
Comment (pro-wh) [reply README.md:95]: hm I don't think we can maintain a copy of the version number here

Comment (pro-wh) [reply README.md:95]: o crud I'm actually not even on that version locally

Comment (pro-wh) [reply README.md:95]: eh I'll add it

File: README.md (hunk 2 above)
Comment (pro-wh) [reply README.md:107]: ideally this and gofumpt similarly should use the same version that golangci-lint uses. but bleh

Comment (pro-wh) [reply README.md:107]: I'll leave the instruction at latest but add a note

**Pull request merged**